docs/concepts/models/inference-parameters.md

```diff
@@ -0,0 +1,147 @@
+
+!!! note "Default Values"
+    If `temperature`, `top_p`, or `max_tokens` are not provided, the model provider's default values will be used. Different providers and models may have different defaults.
+
+!!! tip "Controlling Reasoning Effort for GPT-OSS Models"
+    For gpt-oss models like `gpt-oss-20b` and `gpt-oss-120b`, you can control the reasoning effort using the `extra_body` parameter:
+
+    ```python
+    from data_designer.essentials import ChatCompletionInferenceParameters
+
+    # High reasoning effort (more thorough, slower)
+    inference_parameters = ChatCompletionInferenceParameters(
+        extra_body={"reasoning_effort": "high"}
+    )
+
+    # Medium reasoning effort (balanced)
+    inference_parameters = ChatCompletionInferenceParameters(
+        extra_body={"reasoning_effort": "medium"}
+    )
+
+    # Low reasoning effort (faster, less thorough)
+    inference_parameters = ChatCompletionInferenceParameters(
+        extra_body={"reasoning_effort": "low"}
+    )
+    ```
+
+### Temperature and Top P Guidelines
+
+- **Temperature**:
+    - `0.0-0.3`: Highly deterministic, focused outputs (ideal for structured/reasoning tasks)
+    - `0.4-0.7`: Balanced creativity and coherence (general purpose)
+    - `0.8-1.0`: Creative, diverse outputs (ideal for creative writing)
+    - `1.0+`: Highly random and experimental
+
+- **Top P**:
+    - `0.1-0.5`: Very focused, only most likely tokens
+    - `0.6-0.9`: Balanced diversity
+    - `0.95-1.0`: Maximum diversity, including less likely tokens
+
+!!! tip "Adjusting Temperature and Top P Together"
+    When tuning both parameters simultaneously, consider these combinations:
+
+    - **For deterministic/structured outputs**: Low temperature (`0.0-0.3`) + moderate-to-high top_p (`0.8-0.95`)
+        - The low temperature ensures focus, while top_p allows some token diversity
+    - **For balanced generation**: Moderate temperature (`0.5-0.7`) + high top_p (`0.9-0.95`)
+        - This is a good starting point for most use cases
+    - **For creative outputs**: Higher temperature (`0.8-1.0`) + high top_p (`0.95-1.0`)
+        - Both parameters work together to maximize diversity
+
+    **Avoid**: Setting both very low (overly restrictive) or adjusting both dramatically at once. When experimenting, adjust one parameter at a time to understand its individual effect.
+
+## Distribution-Based Inference Parameters
+
+For `temperature` and `top_p` in `ChatCompletionInferenceParameters`, you can specify distributions instead of fixed values. This allows Data Designer to sample different values for each generation request, introducing controlled variability into your synthetic data.
+
+### Uniform Distribution
+
+Samples values uniformly between a low and high bound:
+
+```python
+from data_designer.essentials import (
+    ChatCompletionInferenceParameters,
+    UniformDistribution,
+    UniformDistributionParams,
```

Comment:
Since we a going to deprecate `InferenceParameters`, maybe we should follow the `SamplerParams` and `DistributionParams` and use `ChatCompletionInferenceParams`, `EmbeddingInferenceParams`, etc?